pluginManager/src/PluginList.cpp

```diff
@@ -904,18 +904,21 @@ void PluginList::downloadList()
 	}
 	else
 	{
-		downloadResult = downloadManager.getUrl(PLUGINS_MD5_URL, serverMD5, &g_options.moduleInfo);
+		TCHAR *md5Url = getPluginsMd5Url();
+		downloadResult = downloadManager.getUrl(md5Url, serverMD5, &g_options.moduleInfo);
 	}
 #else
-	// OSes less than vista don't support SNI, which cloudflare uses to support HTTPS, so we have to use HTTP on old OSes
-    TCHAR *md5Url = (g_options.forceHttp || g_winVer < WV_VISTA) ? PLUGINS_HTTP_MD5_URL : PLUGINS_MD5_URL;
+	TCHAR *md5Url = getPluginsMd5Url();
 	BOOL downloadResult = downloadManager.getUrl(md5Url, serverMD5, &g_options.moduleInfo);
 #endif
 
 	std::shared_ptr<char> cHashBuffer = WcharMbcsConverter::tchar2char(hashBuffer);
 
-	
-	if (downloadResult && serverMD5 != cHashBuffer.get())
+	if (downloadResult && serverMD5 == cHashBuffer.get()) {
+		// Server hash matches local hash, so we're ok to continue
+		downloadSuccess = TRUE;

```

Comment:
This is the actual fix. If the hash matched with the server, it would error later.